File: backend-dotnet/Models/Fund.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace StockPredictApi.Models;

[Table("funds")]
public class Fund
{
    [Key]
    [Column("fund_code")]
    [StringLength(6)]
    public string FundCode { get; set; } = string.Empty;

    [Column("fund_name")]
    [StringLength(100)]
    [Required]
    public string FundName { get; set; } = string.Empty;

    [Column("fund_type")]
    [StringLength(20)]
    [Required]
    public string FundType { get; set; } = string.Empty;

    [Column("company")]
    [StringLength(100)]
    public string? Company { get; set; }

    [Column("manager")]
    [StringLength(100)]
    public string? Manager { get; set; }

    [Column("inception_date")]
    [StringLength(10)]
    public string? InceptionDate { get; set; }

    [Column("latest_nav")]
    public double LatestNav { get; set; } = 0.0;

    [Column("cumulative_nav")]
    public double CumulativeNav { get; set; } = 0.0;

    [Column("return_1m")]
    public double Return1M { get; set; } = 0.0;

    [Column("return_3m")]
    public double Return3M { get; set; } = 0.0;

    [Column("return_6m")]
    public double Return6M { get; set; } = 0.0;

    [Column("return_1y")]
    public double Return1Y { get; set; } = 0.0;

    [Column("return_3y")]
    public double Return3Y { get; set; } = 0.0;

    [Column("risk_level")]
    [StringLength(10)]
    public string? RiskLevel { get; set; }

    [Column("strategy")]
    [StringLength(500)]
    public string? Strategy { get; set; }

    [Column("estimated_nav")]
    public double EstimatedNav { get; set; } = 0.0;

    [Column("change_pct")]
    public double ChangePct { get; set; } = 0.0;

    [Column("updated_at")]
    public DateTime? UpdatedAt { get; set; }
}

File: frontend/src/types/fund.ts
/** Fund search result item. */
export interface FundItem {
  /** Fund code, six digits. */
  fund_code: string
  /** Fund display name. */
  fund_name: string
  /** Fund category, such as mixed, index, or stock fund. */
  fund_type: string
  /** Fund company. */
  company?: string
  /** Fund manager. */
  manager?: string
  /** Latest NAV. */
  latest_nav?: number
  /** Accumulated NAV. */
  cumulative_nav?: number
  /** Estimated NAV. */
  estimated_nav?: number
  /** Daily percentage change. */
  change_pct?: number
  /** One-month return. */
  return_1m?: number
  /** Three-month return. */
  return_3m?: number
  /** Six-month return. */
  return_6m?: number
  /** One-year return. */
  return_1y?: number
  /** Three-year return. */
  return_3y?: number
  /** Risk level. */
  risk_level?: string
  /** Fund inception date. */
  inception_date?: string
  /** Quote date or estimated quote time. */
  quote_date?: string
  /** Quote data source. */
  quote_source?: string
}

export interface FundSearchData {
  items: FundItem[]
  total: number
  page: number
  size: number
}

export interface FundFilters {
  types: string[]
  companies: string[]
  risk_levels: string[]
}
